ihmc-communication/src/main/java/us/ihmc/communication/ros2/sync/ROS2PeerClockOffsetEstimator.java

```diff
@@ -0,0 +1,143 @@
+package us.ihmc.communication.ros2.sync;
+
+import ihmc_common_msgs.msg.dds.PeerClockOffsetEstimatorPingMessage;
+import us.ihmc.commons.exception.DefaultExceptionHandler;
+import us.ihmc.commons.thread.RepeatingTaskThread;
+import us.ihmc.commons.thread.ThreadTools;
+import us.ihmc.communication.ROS2Tools;
+import us.ihmc.communication.packets.MessageTools;
+import us.ihmc.pubsub.common.Guid;
+import us.ihmc.pubsub.common.SampleInfo;
+import us.ihmc.ros2.ROS2Node;
+import us.ihmc.ros2.ROS2Publisher;
+import us.ihmc.ros2.ROS2Topic;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Sends out pings to other instances of this class on the network to estimate
+ * the clock offsets on those computers. This can be used to synchronize data
+ * modifications. We account for and assume a symmetric network delay.
+ */
+public class ROS2PeerClockOffsetEstimator
```

Comment:
What about making this class a singleton? I can't imagine why you'd want more than one instance per process.

Reply:
nah I don't think it should be

Reply:
That would go for a lot of things, I'd rather not go there on all of them